src/main/java/de/hysky/skyblocker/skyblock/events/EventNotifications.java

```diff
@@ -100,35 +143,50 @@ public static void refreshEvents() {
             }
             return List.<JsonObject>of();
-        }).thenAccept(eventsList -> {
-            events.clear();
-            for (JsonObject object : eventsList) {
-                if (object.get("timestamp").getAsLong() + object.get("duration").getAsInt() < currentTime) continue;
-                SkyblockEvent skyblockEvent = SkyblockEvent.of(object);
-                events.computeIfAbsent(object.get("event").getAsString(), s -> new LinkedList<>()).add(skyblockEvent);
-            }
+        })
+        .thenAccept(objects -> {
+            long nowSeconds = System.currentTimeMillis() / 1000;
+            Map<String, LinkedList<SkyblockEvent>> newEvents = objects.stream()
+                    .map(SkyblockEvent::of)
+                    .filter(event -> event.start + event.duration >= nowSeconds)
+                    .collect(Collectors.groupingBy(SkyblockEvent::name, Collectors.toCollection(LinkedList::new)));
+
+            addFixedEvents(nowSeconds, newEvents);
 
-            for (Map.Entry<String, LinkedList<SkyblockEvent>> entry : events.entrySet()) {
+            for (Map.Entry<String, LinkedList<SkyblockEvent>> entry : newEvents.entrySet()) {
                 entry.getValue().sort(Comparator.comparingLong(SkyblockEvent::start)); // Sort just in case it's not in order for some reason in API
                 //LOGGER.info("Next {} is at {}", entry.getKey(), entry.getValue().peekFirst());
             }
 
-            for (String s : events.keySet()) {
-                SkyblockerConfigManager.get().eventNotifications.eventsReminderTimes.computeIfAbsent(s, s1 -> DEFAULT_REMINDERS);
+            for (String s : newEvents.keySet()) {
+                SkyblockerConfigManager.get().eventNotifications.eventsReminderTimes.computeIfAbsent(s, EventNotifications::getDefaultReminders);
             }
+
+            events.set(newEvents);
         }).exceptionally(EventNotifications::itBorked);
     }
 
+    private static void addFixedEvents(long nowSeconds, Map<String, LinkedList<SkyblockEvent>> events) {
+        PERIODIC_EVENTS.forEach(event -> {
+            long firstEvent = SkyblockTime.SKYBLOCK_EPOCH / 1000 + event.offsetFromSkyblockEpochSeconds;
+            long currentEvent = firstEvent + (nowSeconds - firstEvent + event.periodSeconds - 1) / event.periodSeconds * event.periodSeconds;
+            // Prepare events for next two update periods in case one is skipped for some reason, or at least one if event is rare.
+            long eventsToAdd = Math.max(1, 2 * EVENTS_UPDATE_PERIOD.toSeconds() / event.periodSeconds);
+            events.put(event.name, LongStream.range(0, eventsToAdd)
+                    .map(i -> currentEvent + i * event.periodSeconds)
+                    .mapToObj(ts -> new SkyblockEvent(event.name, ts, event.durationSeconds, new String[0], event.warpCommand))
+                    .collect(Collectors.toCollection(LinkedList::new)));
+        });
+    }
```

Comment:
The lambda (passed into `forEach`) could be turned into a method in `SkyblockerPeriodicEvent`. Also, please add some javadoc for what this method does, as the name `add` could be unclear.